gengokit/httptransport/httptransport.go

```diff
@@ -260,9 +260,19 @@ err = json.Unmarshal([]byte({{.LocalName}}Str), {{.LocalName}})`
 		// All repeated args of any type are represented as slices, and bare
 		// assignments to a slice accept a slice as the rvalue. As a result,
 		// LocalName will be declared as a slice, and json.Unmarshal handles
-		// everything else for us.
+		// everything else for us. Addititionally, if a type is a Base type and
+		// is repeated, we first attempt to unmarshal the string we're
+		// provided, and if that fails, we try to unmarshal the string
+		// surrounded by square brackets. If THAT fails, then the string does
+		// not represent a valid JSON string and an error is returned.
 		repeatedUnmarshalTmpl := `
 var {{.LocalName}} {{.GoType}}
+{{- if and (and .IsBaseType .Repeated) (not (Contains .GoType "[]byte"))}}
+err = json.Unmarshal([]byte({{.LocalName}}Str), &{{.LocalName}})
```

Comment:
@zaquestion @hasIan @hasLeland, 

I now am back to thinking that Lelands original method of unmarshaling twice is the correct response. 

This is because the truss generated client code WILL ALWAYS send json on http transport. Because of this, just unmarshaling it without a check will always work with truss clients. 

We should be optimizing for truss to truss communication. While it would be nice to support this without the `[]`s, we should expect them as truss clients will send them. Therefore just unmashal and then try something else.

Note: This whole pull request was because I wanted to have new truss servers communicate with Lee's truss clients. I have sense realized that this is even less possible, as even with this change every string still requires quotes around it, and Lee's services do not send them with quotes.

So could we merge this as is? Less work for Leland and it will be more efficent in the truss to truss case which should be at least the 90% plus case.